massa-deferred-calls/src/lib.rs

```diff
@@ -0,0 +1,482 @@
+            }
+
+            serialized_call.extend(serialized_value.iter());
+        }
+
+        match self
+            .call_deserializer
+            .deserialize::<DeserializeError>(&serialized_call)
+        {
+            Ok((_rest, call)) => Some(call),
+            Err(_) => None,
+        }
+    }
+
+    /// Returns the total amount of gas booked for a slot
+    pub fn get_slot_gas(&self, slot: &Slot) -> u64 {
+        // By default, if it is absent, it is 0
+        let key = deferred_call_slot_total_gas_key!(slot.to_bytes_key());
+        match self.db.read().get_cf(STATE_CF, key) {
+            Ok(Some(v)) => {
+                let result = self
+                    .call_deserializer
+                    .u64_var_int_deserializer
+                    .deserialize::<DeserializeError>(&v)
+                    .expect(DEFERRED_CALL_DESER_ERROR)
+                    .1;
+                result
+            }
+            _ => 0,
+        }
+    }
+
+    /// Returns the base fee for a slot
+    pub fn get_slot_base_fee(&self, slot: &Slot) -> Amount {
+        let key = deferred_call_slot_base_fee_key!(slot.to_bytes_key());
+        match self.db.read().get_cf(STATE_CF, key) {
+            Ok(Some(v)) => {
+                self.call_deserializer
+                    .amount_deserializer
+                    .deserialize::<DeserializeError>(&v)
+                    .expect(DEFERRED_CALL_DESER_ERROR)
+                    .1
+            }
+            _ => Amount::zero(),
+        }
+    }
+
+    /// Returns the total amount of gas booked
+    pub fn get_total_gas(&self) -> u128 {
+        match self
+            .db
+            .read()
+            .get_cf(STATE_CF, DEFERRED_CALL_TOTAL_GAS.as_bytes().to_vec())
+            .expect(CRUD_ERROR)
+        {
+            Some(v) => {
+                let result = self
+                    .registry_changes_deserializer
+                    .total_gas_deserializer
+                    .deserialize::<DeserializeError>(&v)
+                    .expect(DEFERRED_CALL_DESER_ERROR)
+                    .1;
+                match result {
+                    SetOrKeep::Set(v) => v,
+                    SetOrKeep::Keep => 0,
+                }
+            }
+            None => 0,
+        }
+    }
+
+    pub fn put_entry(
+        &self,
+        slot: &Slot,
+        call_id: &DeferredCallId,
+        call: &DeferredCall,
+        batch: &mut DBBatch,
+    ) {
+        let mut buffer_id = Vec::new();
+        self.call_id_serializer
+            .serialize(call_id, &mut buffer_id)
+            .expect(DEFERRED_CALL_SER_ERROR);
+
+        let slot_bytes = slot.to_bytes_key();
+
+        let db = self.db.read();
+
+        // sender address
+        let mut temp_buffer = Vec::new();
+        self.call_serializer
+            .address_serializer
+            .serialize(&call.sender_address, &mut temp_buffer)
+            .expect(DEFERRED_CALL_SER_ERROR);
+        db.put_or_update_entry_value(
```

Comment:
is it ok to put() when we take only a read() lock on db ?

Reply:
fn definition : 
```rust
  fn put_or_update_entry_value(&self, batch: &mut DBBatch, key: Vec<u8>, value: &[u8]);
``` 
Does not need &mut self and it's same for async-pool and Ledger so i think yes but i asked myself the same question